src/main/java/org/mskcc/limsrest/limsapi/SetOrCreateBanked.java

```diff
@@ -206,9 +211,10 @@ public Object execute(VeloxConnection conn) {
                 bankedFields.put("Assay", assayString);
             }
             if (!"NULL".equals(sampleId)) {
+                bankedFields.put("OtherSampleId", userId);
                 bankedFields.put("UserSampleID", userId);
             }
-            if (!"NULL".equals(investigator)) {
+            if (!"NULL".equals(investigator) && setInvestigator) {

```

Comment:
couldn't be just:
 if (!"NULL".equals(investigator) && "".equals(banked.getStringVal("Investigator", user))) {}
??
for new banked sample it will be always empty and for exisitng one it will be checked whether it's empty or not. Or I don't get some other logic behind it?

Reply:
This is just to deal with how the lims will spit nulls if banked doesn't have an Investigator field.